artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQConsumer.java

```diff
@@ -293,6 +295,7 @@ public int handleDeliver(MessageReference reference, ICoreMessage message) {
          }
          //handleDeliver is performed by an executor (see JBPAPP-6030): any AMQConsumer can share the session.wireFormat()
          dispatch = OpenWireMessageConverter.createMessageDispatch(reference, message, session.wireFormat(), this, session.getCoreServer().getNodeManager().getUUID());
+         dispatch.getMessage().getMessageId().setBrokerSequenceId(deliveredSequenceId.getAndAdd(1));
```

Comment:
The createMessageDispatch method already looks to set this, to the existing different value...perhaps it should be changed to pass this value in there to set, helping avoid any inconsistency later.

Also getAndAdd - > getAndIncrement ?

Reply:
good idea, there will be more impact by that api change but it is better. thanks